File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        temp_1 = 1
        temp_2 = 2
        if n == 1 :
            return temp_1
        elif n == 2 :
            return temp_2
        else :
            for i in range(3,n+1):
                temp = temp_1 + temp_2
                temp_1 = temp_2
                temp_2 = temp
            return temp_2

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int base_1 = 1;
        int base_2 = 2;
        if( n == 1 || n == 2 ) return ( n == 1 ) ? base_1 : base_2;
        else {
            for(int i = 3; i <= n; i++ ){
                int temp = base_1 + base_2;
                base_1 = base_2;
                base_2 = temp;
            }
            return base_2;
        }
    }
};